src/components/AssetsList/index.jsx

```diff
@@ -0,0 +1,221 @@
+            nextProps.selectedAsset),
+        });
+      } else {
+        this.setState({
+          selectedAssetIndex: -1,
+        });
+      }
+    }
+
+    /*
+      if the selectedAsset is set to empty object, reset selectedAssetIndex; this is to
+      handle the case when the EditImageModal closes
+    */
+    if (Object.keys(this.props.selectedAsset).length !== 0 &&
+        Object.keys(nextProps.selectedAsset).length === 0) {
+      this.setState({
+        ...initialState,
+      });
+    }
+  }
+
+  onListBoxFocus = () => {
+    // if no list item is selected, select first item on  listbox focus
+    if (this.state.selectedAssetIndex === -1) {
+      this.setState(
+        { selectedAssetIndex: this.state.selectedAssetIndex + 1,
+          selectedAssetPage: this.props.assetsListMetadata.page,
+        },
+        () => { this.props.selectAsset(this.props.assetsList[this.state.selectedAssetIndex]); },
+      );
+    }
+  }
+
+  onAssetClick = (asset, index) => {
+    this.setState({
+      selectedAssetIndex: index,
+      selectedAssetPage: this.props.assetsListMetadata.page,
+    });
+
+    this.props.selectAsset(asset);
+  }
+
+  getSelectedAssetIndex = (assetsList, selectedAsset) => (assetsList.findIndex(
+    asset => asset.id === selectedAsset.id));
+
+  getImageThumbnail(thumbnail) {
+    const baseUrl = this.props.courseDetails.base_url || '';
+
+    return (
+      <div className={styles['assets-list-image-preview-container']}>
+        {
+          thumbnail ?
+            <img className={styles['assets-list-image-preview-image']} src={`${baseUrl}${thumbnail}`} alt="" data-identifier="asset-image-thumbnail" /> :
+            <WrappedMessage message={messages.assetsListNoPreview} >
+              {displayText => <span className={classNames('text-center')} data-identifier="asset-image-thumbnail">{displayText}</span>}
+            </WrappedMessage>
+        }
+      </div>
+    );
+  }
+
+  getAssetsListHeader = () => (
+    <div className={classNames(styles['list-header'], 'row')}>
+      <span className="col-6 offset-3" data-identifier="asset-file-name" id="name-header"><WrappedMessage message={messages.assetsListNameLabel} /></span>
+      <span className="col-3" data-identifier="asset-date-added" id="date-added-header"><WrappedMessage message={messages.assetsListDateLabel} /></span>
+    </div>
+  )
+
+  getThumbnailElement = thumbnail => (
+    <span aria-hidden className="col">{this.getImageThumbnail(thumbnail)}</span>
+  );
+
+  getDisplayNameElement = (displayName, index) => (
+    <span id={`asset-name-${index}`} className="col-6" data-identifier="asset-file-name">{displayName}</span>
+  );
+
+  getDateAddedElement = (dateAdded, index) => (
+    <span id={`asset-date-${index}`} className="col-3" data-identifier="asset-date-added">{dateAdded}</span>
+  );
+
+  getAssetListItem = (asset, index) => {
+    const isSelected = this.props.selectedAsset.id === asset.id;
+
+    return (<li
```

Comment:
Nit: I think it would look a little better if this `<li` started on a newline below.